### PR TITLE
Optional requirements

### PR DESCRIPTION
**Pull request by gsakkis**
Make the visualization dependencies optional since they pull a bunch of indirect dependencies (especially `tiledb-plot-widget`), which increases the environment footprint and increases the chance of version conflicts.

To preserve the current behavior when installing use `pip install tiledb-cloud[all]==0.7.22` (adjust for latest version).

### CODE REVIEW

**Review by ihnorton — APPROVED** [review]
LGTM but let's get a sign-off from @Shelnutt2 before merging since he set this up.

**Review by thetorpedodog — APPROVED** [review]
fantastic, I'm on board

**Comment by Shelnutt2** [comment]
@gsakkis can you update this PR and provide example update to docs for how this is to be used?

**Comment by gsakkis** [comment]
@Shelnutt2 the README points to https://docs.tiledb.com/cloud/client-api/installation for installation instructions; should I update the docs there?

**Review by Shelnutt2 — APPROVED** [review]

**Comment by Shelnutt2** [comment]
> @Shelnutt2 the README points to https://docs.tiledb.com/cloud/client-api/installation for installation instructions; should I update the docs there?

Yes those, and also a snippet in this PR for us to reference in the next release.

**Comment by gsakkis** [comment]
Updated the description with the equivalent `pip install` command; I'll update the docs separately.

**Comment by gsakkis** [comment]
Should I update or close this?

**Comment by gsakkis** [comment]
> I'm in favor of updating and getting it merged. Then also updating the downstream packages.

Updated; I'll push PRs for the docker images when 0.7.28 is released.

**Pull request merged**
